doc/users/whats_new/path_simplification_updates.rst

```diff
@@ -0,0 +1,8 @@
+Path simplification updates
+---------------------------
+
+Lines will now be simplified according to the ``path.simplify`` and
+``path.simplify_threshold`` RC parameters, as long as other conditions
+such as not using dashed lines are met. Simplification will only happen
+when drawing the line segment portion of a path, if you are also drawing
+markers you should consider using the ``markevery`` option to ``plot``.
```

Comment:
There was some discussion as to folding path.simplify and path.simplify_threshold into a single rcparam (zero vs nonzero) that got lost in the thread, too.  @kbrose Do you plan to do it?  It can be a separate PR of course.